### ISSUE TITLE
Chef backup does not work if it is not /var/chef

### ISSUE THREAD

**Issue by neliel123** [chef/chef#5959]
## Description

I wanted to backup my configuration file to a specific folder directory in my linux workstation prior to the template overwriting it.  I wanted to override the default backup directory by chef
So I tried doing this.  

```
Chef::Config[:file_backup_path] = "/var/my_backup"

template "myconfig.ini" do
  source 'myconfig.ini.erb'
  backup 5
end 
```

It does not work.  Everything is getting back up still at the /var/chef/backup folder.
However changing it to this code, works with no issues.  

```
Chef::Config[:file_backup_path] = "/var/chef/my_backup"

template "myconfig.ini" do
  source 'myconfig.ini.erb'
  backup 5
end 
```
So any folder that is not within the /var/chef directory is not allowed but any folder that is within the /var/chef/ is allowed
## Chef Version

Chef-solo 12.6.0

## Platform Version

Linux

## Replication Case

1.  Create a recipe that creates a template.  A file should be existing on that disk.
2.  On the template resource set the backup property.
3.  Prior to the template being called, set the file_backup_path to a different directory other than the /var/chef/.

## Client Output

Backup does not work.  It is still is backing the directory to /var/chef/backup.
## Stacktrace

No Stacktrace as there is no exception thrown only that the output is not expected

### NOTE: CHEF CLIENT BUGS ONLY

This issue tracker is for the code contained within this repo -- `chef-client`, base `knife` functionality (not
plugins), `chef-apply`, `chef-solo`, `chef-client -z`, etc.

* Requests for new or alternative functionality should be made to [feedback.chef.io](https://feedback.chef.io/forums/301644-chef-product-feedback/category/110832-chef-client)
* [Chef Server issues](https://github.com/chef/chef-server/issues/new)
* [ChefDK issues](https://github.com/chef/chef-dk/issues/new)
* Cookbook Issues (see the https://github.com/chef-cookbooks repos or search [Supermarket](https://supermarket.chef.io) or GitHub/Google)

**Comment by thommay** [comment]
Does `/var/my_backup` exist?

**Comment by neliel123** [comment]
Yes it exist prior to me running the code.  I am running with root privilege also.

**Comment by thommay** [comment]
Interesting. Can you try with an up to date chef client, please? And then gist a debug log of the run with `chef-solo -l debug` and put the link here.

**Comment by neliel123** [comment]
I did a quick yum remove of my chef..then I downloaded my rpm.  My chef-solo version is now
```
chef-solo -v
Chef: 12.19.36
```
Retried my code but still the same.  Putting it in subdirectory of /var/chef works though.  But other than that, it is ignored.

Sorry but I am new to chef and I am just trying it out..I am not sure howto give you the logs.  I tried running debug but I am getting some exception
